### ISSUE TITLE
错误请教

### ISSUE THREAD

**Issue by MapleFigure** [takaniwa/DSNet#4]
![屏幕截图 2024-06-19 213315](https://github.com/takaniwa/DSNet/assets/99528033/0d1c8c63-5f39-4a06-83c6-4000bbdddc61)
请问一下，这是什么原因造成的

**Comment by takaniwa** [comment]
> 请问一下train.lst文件怎么获取

感谢你的热情,请稍等,我忘了上传data list了

**Comment by takaniwa** [comment]
> 请问一下train.lst文件怎么获取

我已经上传 lst文件,数据集存储格式可以参考lst文件

**Comment by MapleFigure** [comment]
> > 请问一下train.lst文件怎么获取
> 
> 我已经上传 lst文件,数据集存储格式可以参考lst文件

谢谢

**Comment by MapleFigure** [comment]
def _ohem_forward(self, score, target, **kwargs):
        pred = F.softmax(score, dim=1)
        pixel_losses = self.criterion(score, target).contiguous().view(-1)
        mask = target.contiguous().view(-1) != self.ignore_label
        tmp_target = target.clone()
        tmp_target[tmp_target == self.ignore_label] = 0
        pred = pred.gather(1, tmp_target.unsqueeze(1))
        pred, ind = pred.contiguous().view(-1, )[mask].contiguous().sort()
        min_value = pred[min(self.min_kept, pred.numel() - 1)]
        threshold = max(min_value, self.thresh)
        pixel_losses = pixel_losses[mask][ind]
        pixel_losses = pixel_losses[pred < threshold]
        return pixel_losses.mean()
![屏幕截图 2024-06-20 215437](https://github.com/takaniwa/DSNet/assets/99528033/667e024f-2afb-4c60-95fe-ec593bbe0974)
请教一下这个错误是什么原因造成的

**Comment by takaniwa** [comment]
> ```
> def _ohem_forward(self, score, target, **kwargs):
>     pred = F.softmax(score, dim=1)
>     pixel_losses = self.criterion(score, target).contiguous().view(-1)
>     mask = target.contiguous().view(-1) != self.ignore_label
>     tmp_target = target.clone()
>     tmp_target[tmp_target == self.ignore_label] = 0
>     pred = pred.gather(1, tmp_target.unsqueeze(1))
>     pred, ind = pred.contiguous().view(-1, )[mask].contiguous().sort()
>     min_value = pred[min(self.min_kept, pred.numel() - 1)]
>     threshold = max(min_value, self.thresh)
>     pixel_losses = pixel_losses[mask][ind]
>     pixel_losses = pixel_losses[pred < threshold]
>     return pixel_losses.mean()
> ```
> 
> ![屏幕截图 2024-06-20 215437](https://private-user-images.githubusercontent.com/99528033/341438862-667e024f-2afb-4c60-95fe-ec593bbe0974.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4OTU2NDcsIm5iZiI6MTcxODg5NTM0NywicGF0aCI6Ii85OTUyODAzMy8zNDE0Mzg4NjItNjY3ZTAyNGYtMmFmYi00YzYwLTk1ZmUtZWM1OTNiYmUwOTc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIwVDE0NTU0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0MjdjMGYyOTk1NmIyM2U3YmRmZWM0NzAyMTlmOWEzZTMzZTNhMmNkYTM5NmNjZmUyNDVmZTdmNTFmODQzYzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.p7aoGtdrJzF9V9eAZkzylVkTteW_1ESS5LSUTOddhdY) 请教一下这个错误是什么原因造成的

不好意思，等几天吧，我尽量在下周之前把脚本改好，这边的代码都是手动创建上传的，可能会有一些错误。改好了我会给你留言

**Comment by takaniwa** [comment]
> ```
> def _ohem_forward(self, score, target, **kwargs):
>     pred = F.softmax(score, dim=1)
>     pixel_losses = self.criterion(score, target).contiguous().view(-1)
>     mask = target.contiguous().view(-1) != self.ignore_label
>     tmp_target = target.clone()
>     tmp_target[tmp_target == self.ignore_label] = 0
>     pred = pred.gather(1, tmp_target.unsqueeze(1))
>     pred, ind = pred.contiguous().view(-1, )[mask].contiguous().sort()
>     min_value = pred[min(self.min_kept, pred.numel() - 1)]
>     threshold = max(min_value, self.thresh)
>     pixel_losses = pixel_losses[mask][ind]
>     pixel_losses = pixel_losses[pred < threshold]
>     return pixel_losses.mean()
> ```
> 
> ![屏幕截图 2024-06-20 215437](https://private-user-images.githubusercontent.com/99528033/341438862-667e024f-2afb-4c60-95fe-ec593bbe0974.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzNjYzMzUsIm5iZiI6MTcxOTM2NjAzNSwicGF0aCI6Ii85OTUyODAzMy8zNDE0Mzg4NjItNjY3ZTAyNGYtMmFmYi00YzYwLTk1ZmUtZWM1OTNiYmUwOTc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI2VDAxNDAzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1NjJlYzMyZDlmNzJmODVjNWEwNjAyYTc5ZjkzN2JjZjY4ZmM0OTI2YTczYTJkMWQ5ZjNmNGI1MGRkZGUwZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gziL7RGK5y44_hbmcpjn2bHIUZvpdOcgrSigji

**Comment by saddogs** [comment]
你好，这个链接一直进不去，请问还有其他下载该文件的方法吗
![22222222](https://github.com/takaniwa/DSNet/assets/145547433/c4b3aa92-45c1-4fcb-bd1a-03b11360c900)
